uint32 layer_level
---
uint32 layer_size
